halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1545,6 +1685,472 @@ impl<F: Field> Gate<F> {
+            cs: cs2,
+            preprocessing,
+        },
+        config,
+        cs,
+    ))
+}
+
+impl<F: Field> ConstraintSystemV2Backend<F> {
+    pub(crate) fn collect_queries(
+        &self,
+    ) -> (
+        Queries,
+        Vec<Gate<F>>,
+        Vec<lookup::Argument<F>>,
+        Vec<shuffle::Argument<F>>,
+    ) {
+        let mut queries = QueriesMap {
+            advice_map: HashMap::new(),
+            instance_map: HashMap::new(),
+            fixed_map: HashMap::new(),
+            advice: Vec::new(),
+            instance: Vec::new(),
+            fixed: Vec::new(),
+        };
+
+        let gates: Vec<_> = self
+            .gates
+            .iter()
+            .map(|gate| Gate {
+                name: gate.name.clone(),
+                constraint_names: gate.constraint_names.clone(),
+                polys: gate
+                    .polynomials()
+                    .iter()
+                    .map(|e| queries.as_expression(e))
+                    .collect(),
+                queried_selectors: Vec::new(), // Unused?
+                queried_cells: Vec::new(),     // Unused?
+            })
+            .collect();
+        let lookups: Vec<_> = self
+            .lookups
+            .iter()
+            .map(|lookup| lookup::Argument {
+                name: lookup.name.clone(),
+                input_expressions: lookup
+                    .input_expressions
+                    .iter()
+                    .map(|e| queries.as_expression(e))
+                    .collect(),
+                table_expressions: lookup
+                    .table_expressions
+                    .iter()
+                    .map(|e| queries.as_expression(e))
+                    .collect(),
+            })
+            .collect();
+        let shuffles: Vec<_> = self
+            .shuffles
+            .iter()
+            .map(|shuffle| shuffle::Argument {
+                name: shuffle.name.clone(),
+                input_expressions: shuffle
+                    .input_expressions
+                    .iter()
+                    .map(|e| queries.as_expression(e))
+                    .collect(),
+                shuffle_expressions: shuffle
+                    .shuffle_expressions
+                    .iter()
+                    .map(|e| queries.as_expression(e))
+                    .collect(),
+            })
+            .collect();
+
+        for column in self.permutation.get_columns() {
+            match column.column_type {
+                Any::Instance => {
+                    queries.add_instance(Column::new(column.index(), Instance), Rotation::cur())
+                }
+                Any::Fixed => {
+                    queries.add_fixed(Column::new(column.index(), Fixed), Rotation::cur())
+                }
+                Any::Advice(advice) => {
+                    queries.add_advice(Column::new(column.index(), advice), Rotation::cur())
+                }
+            };
```

Comment:
ditto

Reply:
After https://github.com/privacy-scaling-explorations/halo2/pull/243/commits/20c16ddf4272f1e92decc514a55d48aafbf0cea0 the function `collect_queries` is much smaller.  I think it's not worth moving this to a new function because this is already a small chunk of code; please take a look at how it looks now and let me know what you think: https://github.com/privacy-scaling-explorations/halo2/blob/20c16ddf4272f1e92decc514a55d48aafbf0cea0/halo2_proofs/src/plonk/circuit.rs#L2124

Reply:
Fair! Makes sense though!